components/brave_rewards/browser/safetynet_check.cc

```diff
@@ -0,0 +1,78 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/safetynet_check.h"
+
+#include <memory>
+#include <utility>
+
+#include "base/android/jni_android.h"
+#include "base/android/jni_string.h"
+#include "brave/build/android/jni_headers/SafetyNetCheck_jni.h"
```

Comment:
this one also needs to go into `components/brave_rewards/browser` because it needs a dep for the jni_headers, but is not allowed to have a dep on `//brave/build`. I think you just need to add another `generate_jni("jni_headers")` target for `components/brave_rewards/browser` for SafetyNetCheck.java